src/docfx/restore/Restore.cs

```diff
@@ -77,16 +126,39 @@ public static (string restoreDir, string url, string refSpec) GetGitRestoreInfo(
                     if (GitUtility.IsRepo(restoreDir))
                     {
                         // already exists, just pull the new updates from remote
-                        await GitUtility.Pull(restoreDir);
+                        await GitUtility.Fetch(restorePath);
                     }
                     else
                     {
                         // doesn't exist yet, clone this repo to a specified branch
-                        await GitUtility.Clone(restoreDir, url, restoreDir, rev);
+                        await GitUtility.Clone(restoreDir, url, restorePath, null, true);
+                    }
+
+                    workTreeHead = await GitUtility.Revision(restorePath, rev);
+                    await GitUtility.PruneWorkTrees(restorePath);
```

Comment:
>PruneWorkTrees [](start = 37, length = 14)

suggets to execute prune during GC

Reply:
OK, let's remove it and add it back in restore clean PR

---
In reply to: [200885521](https://github.com/dotnet/docfx/pull/3020#discussion_r200885521) [](ancestors = 200885521)